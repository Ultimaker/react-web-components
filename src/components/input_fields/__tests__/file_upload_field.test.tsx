// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import FileUploadField, {FileUploadFieldProps} from '../file_upload_field';
import InputFieldWrapper from '../input_field_wrapper';
import FileUpload from '../../file_upload';

describe('The file upload field component', () => {
    let props: FileUploadFieldProps;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testInputField',
            onChangeHandler: jest.fn(),
            onReadHandler: jest.fn(),
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm',
            staticField: false,
        };
        wrapper = shallow(<FileUploadField {...props} />);
    });

    it('should render a wrapper', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onChangeHandler).not.toHaveBeenCalled();
    });

    it('should render a file upload', () => {
        expect(wrapper.find(FileUpload).props()).toEqual({
            disabled: false,
            id: "testInputField",
            onChangeHandler: wrapper.instance()._onChange,
      });
    })

    it('should call the callback', async () => {
        expect(props.onChangeHandler).not.toHaveBeenCalled();

        const file = new Blob(["A+test+string+for+testing+sources"], {type: 'application/zip'});
        const input = {value: 'c/my/file.zip', files: [file]};
        wrapper.find(FileUpload).prop("onChangeHandler")(input);

        expect(props.onChangeHandler).toHaveBeenCalledWith(props.id, input);

        // wait for file to be read
        await new Promise(setImmediate);
        await new Promise(setImmediate);

        const expected = 'data:application/zip;base64,' + btoa('A+test+string+for+testing+sources');
        expect(props.onReadHandler).toHaveBeenCalledWith(props.id, expected);
        expect(wrapper.find(InputFieldWrapper).prop("touched")).toEqual(true);
    });
});
