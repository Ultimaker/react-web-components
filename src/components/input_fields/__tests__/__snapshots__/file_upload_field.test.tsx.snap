// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The file upload field component should render a wrapper 1`] = `
<InputFieldWrapper
  id="testInputField"
  labelLayoutWidth="1/1"
  labelWidthBreakpoint="sm"
  onChangeHandler={[MockFunction]}
  onReadHandler={[MockFunction]}
  showValidationError={true}
  staticField={false}
  touched={false}
>
  <FileUpload
    buttonText="Choose file"
    disabled={false}
    id="testInputField"
    onChangeHandler={[Function]}
    placeholder="a text placeholder"
    uploading={false}
  />
</InputFieldWrapper>
`;
